src/components/icon/icon.tsx

```diff
@@ -0,0 +1,63 @@
+import { Link, To } from "react-router-dom";
+
+import Activity from "../../assets/icons/style_guide/icon-activity.svg";
+import Airplay from "../../assets/icons/style_guide/icon-airplay.svg";
+import AlertCircle from "../../assets/icons/style_guide/icon-alert-circle.svg";
+import AlertOctagon from "../../assets/icons/style_guide/icon-alert-octagon.svg";
+import AlertTriangle from "../../assets/icons/style_guide/icon-alert-triangle.svg";
+import AlignCenter from "../../assets/icons/style_guide/icon-align-center.svg";
+import AlignJustify from "../../assets/icons/style_guide/icon-align-justify.svg";
+import ArrowDownCircle from "../../assets/icons/style_guide/icon-arrow-down-circle.svg";
+import ArrowDownLeft from "../../assets/icons/style_guide/icon-arrow-down-left.svg";
+import ArrowDownRight from "../../assets/icons/style_guide/icon-arrow-down-right.svg";
+import ArrowDown from "../../assets/icons/style_guide/icon-arrow-down.svg";
+import ArrowLeftCircle from "../../assets/icons/style_guide/icon-arrow-left-circle.svg";
+import ArrowLeft from "../../assets/icons/style_guide/icon-arrow-left.svg";
+import ArrowRightCircle from "../../assets/icons/style_guide/icon-arrow-right-circle.svg";
+import ArrowRight from "../../assets/icons/style_guide/icon-arrow-right.svg";
+import ChevronLeft from "../../assets/icons/style_guide/icon-chevron-left.svg";
+import ChevronRight from "../../assets/icons/style_guide/icon-chevron-right.svg";
+import File from "../../assets/icons/style_guide/icon-file.svg";
+import Logout from "../../assets/icons/style_guide/icon-log-out.svg";
+import Trash from "../../assets/icons/style_guide/icon-trash.svg";
+import Search from "../../assets/icons/style_guide/icon-search.svg";
+import User from "../../assets/icons/style_guide/icon-user.svg";
+import X from "../../assets/icons/style_guide/icon-x.svg";
+import XCircle from "../../assets/icons/style_guide/icon-x-circle.svg";
+
+interface IconProps {
+    className: string;
+    iconType: string;
+    to: To;
+}
+
+export const Icon = (props: IconProps) => {
```

Comment:
Aqui eu tava usando como padrão descrever mesmo os props, eu acho mais fácil de controlar. Mas deixei como comentário pq n é 100% necessário, seria mais pra padronizar o cód mesmo.

Caso queira, seria trocar `props: IconProps` por `{ className, iconType, to }: IconProps`